### PR TITLE
Expiry time - paypro

### LINKED ISSUE #2456: Check expired TX from paypro requests
and auto reject them?

### PR DESCRIPTION
**Pull request by JDonadio**
Fixes #2456

### CODE REVIEW

File: src/js/controllers/walletHome.js [hunk 1]
```diff
@@ -27,6 +27,8 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
   this.showScanner = false;
   this.addr = {};
   this.lockedCurrentFeePerKb = null;
+  self.paymentExpired = true;
```
Comment (cmgustavo) [inline src/js/controllers/walletHome.js]: Please, use `self` only wherever necessary. Take a look at this example: https://discourse.codecombat.com/t/javascript-this-vs-self/3143

File: src/js/controllers/walletHome.js [hunk 2]
```diff
@@ -888,6 +911,8 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
 
   this.resetForm = function() {
     this.resetError();
+    self.paymentExpired = true;
+    self.usePaypro = false;
```
Comment (cmgustavo) [inline src/js/controllers/walletHome.js]: Same as above.

File: public/views/walletHome.html [hunk 3]
```diff
@@ -390,6 +390,7 @@ <h4 class="title m0">
             ng-click="home.sendAll()"
              translate> Send All
           </a>
+          <time ng-show="!home.paymentExpired" translate>Payment expires: {{home.timeToExpire * 1000 | amTimeAgo}}</time>
```
Comment (cmgustavo) [inline public/views/walletHome.html]: Use only the string `Payment expires` to translate.

File: src/js/controllers/walletHome.js [hunk 4]
```diff
@@ -282,6 +287,21 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
       $scope.getShortNetworkName = function() {
         return fc.credentials.networkName.substring(0, 4);
       };
+
+      function checkPaypro() {
+        if(tx.payProUrl && !isChromeApp){
+          fc.fetchPayPro({
+            payProUrl: tx.payProUrl,
+          }, function(err, paypro) {
+            if (err) return $log.debug(err);
```
Comment (cmgustavo) [inline src/js/controllers/walletHome.js]: Should return `null`, not `$log`.

**Comment by tanojaja** [comment]
Updated

**Comment by cmgustavo** [comment]
When opening the transaction details, I can see (for example) `Expires: 20 minutes ago`. I think that It should notify in red text that the invoice expired

**Comment by cmgustavo** [comment]
Need rebase

**Comment by JDonadio** [comment]
Updated.

File: src/js/controllers/walletHome.js [hunk 5]
```diff
@@ -1059,11 +1084,39 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
...
         self._paypro = paypro;
         self.setForm(paypro.toAddress, (paypro.amount * satToUnit).toFixed(self.unitDecimals), paypro.memo);
+        _paymentTimeControl(paypro.expires);
         return cb();
       });
     }, 1);
   };
 
+  function _paymentTimeControl(timeToExpire) {
+    self.usePaypro = true;
```
Comment (matiu) [inline src/js/controllers/walletHome.js]: dont use an other variable. Use: `_paypro`.

File: src/js/controllers/walletHome.js [hunk 6]
```diff
@@ -947,6 +970,8 @@ angular.module('copayApp.controllers').controller('walletHomeController', functi
 
   this.resetForm = function() {
     this.resetError();
+    this.paymentExpired = true;
```
Comment (matiu) [inline src/js/controllers/walletHome.js]: paymentExpired should default to false.

**Comment by JDonadio** [comment]
Updated.

**Comment by cmgustavo** [comment]
ACK

**Pull request merged**